CONTRIBUTING.md

```diff
@@ -1,52 +1,68 @@
-- Scenario where the issue was seen to occur
-- If the issue is with storage, include maya version, maya osh-status and maya omm-status.
-- Errors and log messages that are thrown by the software
-
+When raising issues, please specify the following:
+- Setup details (like hyperconverged/dedicated), orchestration engine - Kubernetes, Docker Swarm etc. 
+- A Scenario where the issue occurred (with details on how to reproduce it)
+- If the issue is with storage, include maya version, maya osh-status, and maya omm-status.
+- Errors and log messages that are displayed by the software.
 
 ## Submit Change to Improve Documentation
 
-Getting Documentation Right is Hard! Please raise a PR with you proposed changes. 
+Getting documentation right is hard! Refer to this [page](./contribute/CONTRIBUTING-TO-DEVELOPER-DOC.md) for more information on how you could improve the developer documentation by submitting pull requests with appropriate tags. Here's a [list of tags](./contribute/labels-of-issues.md) that could be used for the same. Help us keep our documentation clean, easy to understand, and accessible.
 
 ## Submit Proposals for New Features
 
-There is always something more that is required, to make it easier to suit your use-cases. Feel free to join the discussion on the features or raise a new PR with your proposed change. 
+There is always something more that is required, to make it easier to suit your use-cases. Feel free to join the discussion on new features or raise a PR with your proposed change. 
+
+- Join us at [Slack](https://openebsslacksignup.herokuapp.com/)
+ 	 - Already signed up? Head to our discussions at [#openebs-users](https://openebs-community.slack.com/messages/openebs-users/)
 
-## Contributing to Source Code
+## Contributing to Source Code & Bug Fixes
 
-Provide PRs for bug fixes or enhancements.
+Provide PRs with appropriate tags for bug fixes or enhancements to the source code. For a list of tags that could be used, see [this]((./contribute/labels-of-issues.md)).
 
+* For contributing to K8s demo, please refer to this [document](./contribute/CONTRIBUTING-TO-K8S-DEMO.md).
+	- For checking out how OpenEBS works with K8s, refer to this [document](./k8s/README.md) 
+- For contributing to Kubernetes OpenEBS Provisioner, please refer to this [document](./contribute/CONTRIBUTING-TO-KUBERNETES-OPENEBS-PROVISIONER.md).
+* For contributing to CI and e2e, refer to this [document](./contribute/CONTRIBUTING-TO-CI-AND-E2E.md)
+	- Refer to this [overview](./e2e/README.md) of e2e.
+	
+Refer to this [document](./contribute/design/code-structuring.md) for more information on code structuring and guidelines to follow on the same.
+
+## Solve Existing Issues
+Head over to [issues](www.github.com/openebs/openebs/issues) to find issues where help is needed from contributors. See our [list of labels guide](./contribute/labels-of-issues.md) to help you find issues that you can solve faster.
```

Comment:
Link to issues should be: https://github.com/openebs/openebs/issues

Reply:
Right on. Done.